src/main/java/org/isf/menu/manager/UserBrowsingManager.java

```diff
@@ -129,38 +129,75 @@ public boolean deleteUser(User user) throws OHServiceException {
 		return ioOperations.deleteUser(user);
 	}
 
-
+	/**
+	 * Increase the number of failed logon attemptes for {@link User}.
+	 *
+	 * @param user the {@link User}
+	 */
 	public void increaseFailedAttempts(User user) {
 		int newFailAttempts = user.getFailedAttempts() + 1;
 		ioOperations.updateFailedAttempts(user.getUserName(), newFailAttempts);
 	}
 
+	/**
+	 * Reset the number of failed logon attemptes to zero for {@link User}.
+	 *
+	 * @param user the {@link User}
+	 */
 	public void resetFailedAttempts(User user) {
 		ioOperations.updateFailedAttempts(user.getUserName(), 0);
 	}
 
+	/**
+	 * Lock the {@link User} from logging into the system.
+	 *
+	 * @param user the {@link User}
+	 */
 	public void lockUser(User user) throws OHServiceException {
 		user.setAccountLocked(true);
 		user.setLockedTime(TimeTools.getNow());
 		ioOperations.updateUserLocked(user.getUserName(), true, user.getLockedTime());
 	}
 
+	/**
+	 * Unlock the {@link User} so they can log into the system.
+	 *
+	 * @param user the {@link User}
+	 */
+	public void setLastLogin(User user) throws OHServiceException {
+		ioOperations.setLastLogin(user.getUserName(), TimeTools.getNow());
+	}
+
+	/**
+	 * Unlock the {@link User} so they can log into the system.
+	 *
+	 * @param user the {@link User}
+	 */
 	public void unlockUser(User user) throws OHServiceException {
 		user.setAccountLocked(false);
 		user.setLockedTime(null);
 		user.setFailedAttempts(0);
-		ioOperations.updateFailedAttempts(user.getUserName(), 0);
-		ioOperations.updateUserLocked(user.getUserName(), false, null);
+		String userName = user.getUserName();
+		ioOperations.updateFailedAttempts(userName, 0);
+		ioOperations.updateUserLocked(userName, false, null);
+		ioOperations.setLastLogin(userName, null);
```

Comment:
wondering if we can just persist "user" instead to have 3 different calls to service layer?

Reply:
I thought about that but I was trying to make the minimum number of changes

Reply:
ok, maybe we can put a TODO comment and open a new issue (task type) for that, but I didn't make a complete analysis to suggest the content

Reply:
With the number of columns currently in the table the UPDATE command will have a long list of "sets".   Are we really sure we want that given how infrequently most these commands are called?  I why would we want to update all the columns just to set the last login date/time given that particular call *will* be called often.

Reply:
well, this happens for all entities, it is JPA that carries on the fact to update one or more columns, from service point of view it will be only a "entity.save()" method and the JPA repository does the rest.
It is allowed to do differently and it is not wrong though, but it looks to me an anti-pattern for very small entities as pointed out here https://www.baeldung.com/spring-data-partial-update#custom-query.
That's why I'm suggest to just leave a comment to explain WHY it is done differently from other entities.